pkg/gofr/datasource/file/ftp/fs.go

```diff
@@ -0,0 +1,353 @@
+
+// ftpFileSystem represents a file system interface over FTP.
+type ftpFileSystem struct {
+	*ftpFile
+	conn    ServerConn
+	config  *Config
+	logger  Logger
+	metrics Metrics
+}
+
+// Config represents the FTP configuration.
+type Config struct {
+	Host      string // FTP server hostname
+	User      string // FTP username
+	Password  string // FTP password
+	Port      int    // FTP port
+	RemoteDir string // Remote directory path. Base Path for all FTP Operations.
+}
+
+// New initializes a new instance of ftpFileSystem with provided configuration.
+func New(config *Config) FileSystemProvider {
+	return &ftpFileSystem{config: config}
+}
+
+// UseLogger sets the logger interface for the FTP file system.
+func (f *ftpFileSystem) UseLogger(logger interface{}) {
+	if l, ok := logger.(Logger); ok {
+		f.logger = l
+	}
+}
+
+// UseMetrics sets the metrics for the MongoDB client which asserts the Metrics interface.
+func (f *ftpFileSystem) UseMetrics(metrics interface{}) {
+	if m, ok := metrics.(Metrics); ok {
+		f.metrics = m
+	}
+}
+
+// Connect establishes a connection to the FTP server and logs in.
+func (f *ftpFileSystem) Connect() {
+	ftpServer := fmt.Sprintf("%v:%v", f.config.Host, f.config.Port)
+
+	defer f.processLog(&FileLog{Operation: "Connect", Location: ftpServer}, time.Now())
+
+	const dialTimeout = 5 * time.Second
+
+	conn, err := ftp.Dial(ftpServer, ftp.DialWithTimeout(dialTimeout))
+	if err != nil {
+		f.logger.Errorf("Connection failed : %v", err)
+		return
+	}
+
+	f.conn = &Conn{conn}
+
+	f.logger.Logf("Connected to FTP Server : %v", ftpServer)
+
+	err = conn.Login(f.config.User, f.config.Password)
+	if err != nil {
+		f.logger.Errorf("Login failed : %v", err)
+		return
+	}
+
+	f.logger.Logf("Login Successful. Current remote location : %q", f.config.RemoteDir)
+}
+
+// Create creates an empty file on the FTP server.
+func (f *ftpFileSystem) Create(name string) (File, error) {
+	filePath := path.Join(f.config.RemoteDir, name)
+
+	defer f.processLog(&FileLog{Operation: "Create", Location: filePath}, time.Now())
+
+	if name == "" {
+		f.logger.Errorf("Create_File failed. Provide a valid filename : %v", errEmptyFilename)
+		return nil, errEmptyFilename
+	}
+
+	emptyReader := new(bytes.Buffer)
+
+	err := f.conn.Stor(filePath, emptyReader)
+	if err != nil {
+		f.logger.Errorf("Create_File failed. Error creating file with path %q : %v", filePath, err)
+		return nil, err
+	}
+
+	_, s := path.Split(filePath)
+
+	res, err := f.conn.Retr(filePath)
+	if err != nil {
+		f.logger.Errorf("Create_File failed : %v", err)
+		return nil, err
+	}
+
+	f.logger.Logf("Create_File successful. Created file %s at %q", name, filePath)
+
+	defer res.Close()
+
+	return &ftpFile{
+		response: res,
+		name:     s,
+		path:     filePath,
+		conn:     f.conn,
+		logger:   f.logger,
+		metrics:  f.metrics,
+	}, nil
+}
+
+// Mkdir creates a directory on the FTP server. Here, os.FileMode is unused.
+func (f *ftpFileSystem) Mkdir(name string, _ os.FileMode) error {
```

Comment:
Why does this method had os.FileMode in its signature then.

I saw the comment, and I don't get it.

The only explanation I can think about is the fact this method could comply an interface or a func(string, os.FileMode)

But then I would expect a comment about it here.

The current comment explains the code, yes, the file mode param is unused. So the comment only confirm it was done on purpose. But there is no explanation about the why it's like that

Reply:
Added the comment.